tests/utils/test_rest_util.py

```diff
@@ -115,6 +119,165 @@ def test_http_request_with_retry(self, _http_request, sleep):
         self.assertRaises(restutil.HttpError, restutil.http_get,
                           "http://foo.bar")
 
+    @patch("time.sleep")
+    @patch("azurelinuxagent.common.utils.restutil._http_request")
+    def test_http_request_retries_status_codes(self, _http_request, _sleep):
+        _http_request.side_effect = [
+            Mock(status=httpclient.SERVICE_UNAVAILABLE),
+            Mock(status=httpclient.OK)
+        ]
+
+        restutil.http_get("https://foo.bar")
+        self.assertEqual(2, _http_request.call_count)
+        self.assertEqual(1, _sleep.call_count)
+
+    @patch("time.sleep")
+    @patch("azurelinuxagent.common.utils.restutil._http_request")
+    def test_http_request_retries_passed_status_codes(self, _http_request, _sleep):
+        # Ensure the code is not part of the standard set
+        self.assertFalse(httpclient.UNAUTHORIZED in restutil.RETRY_CODES)
+
+        _http_request.side_effect = [
+            Mock(status=httpclient.UNAUTHORIZED),
+            Mock(status=httpclient.OK)
+        ]
+
+        restutil.http_get("https://foo.bar", retry_codes=[httpclient.UNAUTHORIZED])
+        self.assertEqual(2, _http_request.call_count)
+        self.assertEqual(1, _sleep.call_count)
+
+    @patch("time.sleep")
+    @patch("azurelinuxagent.common.utils.restutil._http_request")
+    def test_http_request_raises_for_bad_request(self, _http_request, _sleep):
+        _http_request.side_effect = [
+            Mock(status=httpclient.BAD_REQUEST)
+        ]
+
+        self.assertRaises(BadRequestError, restutil.http_get, "https://foo.bar")
+        self.assertEqual(1, _http_request.call_count)
+
+    @patch("time.sleep")
+    @patch("azurelinuxagent.common.utils.restutil._http_request")
+    def test_http_request_retries_exceptions(self, _http_request, _sleep):
+        # Note:
+        # -- It would be best to test each possible exception,
+        #    but their differing signatures makes that a fair bit of work
```

Comment:
Does it makes sense to assert all of the code you assert based on then?  If the list changes at least you'll break a test, so the breaker knows someone thought about this list previously.